### PR TITLE
fix: always remove message from memory spooler

### PR DESCRIPTION
**Pull request by vabhasin**
**Issue #, if available:**
Given spooler storageType "Disk", and `addMessage` fails for some reason, we add the message to Memory spooler as a fallback. When we call `removeMessageById` it will try to remove message from DiskSpooler with the given ID but message was never added to persistent spooler and instead added to Memory. Due to this, the message will actually never be removed from the memory spooler , which is bad. 
**Description of changes:**
- Always remove message from Memory Spooler.
- Removed a check in `persistentQueueSync` which is never possible now that we moved `persistentQueueSync` back to spooler initialization. Earlier the check was added as we were calling queue sync from Disk Spooler implementation, in that case the condition would have occurred when we restarted the disk Spooler component.
**Why is this change necessary:**

**How was this change tested:**
- [ ] Updated or added new unit tests.
- [ ] Updated or added new integration tests.
- [ ] Updated or added new end-to-end tests.
- [ ] If my code makes a remote network call, it was tested with a proxy.

**Any additional information or context required to review the change:**

**Documentation Checklist:**
 - [ ] Updated the README if applicable.

**Compatibility Checklist:**
- [ ] I confirm that the change is backwards compatible.
- [ ] Any modification or deletion of public interfaces does not impact other plugin components.
- [ ] For external library version updates, I have reviewed its change logs and Nucleus does not consume 
  any deprecated method or type.

Refer to [Compatibility Guidelines](/COMPATIBILITY.md) for more information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java [hunk 1]
```diff
@@ -300,10 +304,6 @@ public void persistentQueueSync(Iterable<Long> diskQueueOfIds, CloudMessageSpool
         int numMessages = 0;
         int queueOfMessageIdInitSize = queueOfMessageId.size();
         for (long currentId : diskQueueOfIds) {
-            // Check if Queue of message IDs already contains this ID
-            if (queueOfMessageId.contains(currentId)) {
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java:304]: why is this removed?

**Review by MikeDombo — COMMENTED** [review]
Where is the NPE? That needs to be fixed and these changes don't really look like they're fixing the root cause.

File: src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java (hunk 1 above)
Comment (vabhasin) [reply src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java:304]: I have it in description.

Comment (MikeDombo) [reply src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java:304]: What happens if it ID does exist in the queue and you're wrong? Is that a problem?

**Comment by MikeDombo** [comment]
Fix your PR title. It must conform to conventional commit format.

**Comment by vabhasin** [comment]
> Where is the NPE? That needs to be fixed and these changes don't really look like they're fixing the root cause.

To be clear, there would be SQLException due to NPE would be from DiskSpooler Remove method, we catch that SQLException. The `removeMessageById` from spool wouldn't throw anything. And we should be ok with this scenario, as its just the case where we fallback to adding to Memory.

**Comment by MikeDombo** [comment]
Why is there an NPE in disk spooler? That needs to be fixed

**Comment by vabhasin** [comment]
> Why is there an NPE in disk spooler? That needs to be fixed

Never mind, my mistake. I was using a modified exampleSpooler component in my UAT run from which I had the error. The modified disk spooler had some extra logs and was trying to print message from DiskSpooler from within the remove method. Long story short, no NPE.

**Review by MikeDombo — APPROVED** [review]

**Review by jcosentino11 — APPROVED** [review]
